src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -42,58 +43,84 @@ public boolean contains(Person toCheck) {
      */
     public String getNonUniqueAttributeType(Person toCheck) {
         requireNonNull(toCheck);
-        StringBuilder stringBuilder = new StringBuilder();
+        ArrayList<String> nonUniqueAttributeTypes = new ArrayList<>();
         for (int i = 0; i < internalList.size(); i++) {
             Person otherPerson = internalList.get(i);
             if (otherPerson.isSamePerson(toCheck)) { // Unique attributes already exist in the address book.
                 if (otherPerson.getPhone().equals(toCheck.getPhone())) {
-                    appendPhoneAttributeToString(stringBuilder);
-                }
-                if (otherPerson.getMatriculationNumber().equals(toCheck.getMatriculationNumber())) {
-                    appendMatriculationNumberAttributeToString(stringBuilder);
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 0);
                 }
                 if (otherPerson.getEmail().equals(toCheck.getEmail())) {
-                    appendEmailAttributeToString(stringBuilder);
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 1);
+                }
+                if (otherPerson.getMatriculationNumber().equals(toCheck.getMatriculationNumber())) {
+                    addNonUniqueAttributeType(nonUniqueAttributeTypes, 2);
                 }
             }
         }
-        return stringBuilder.toString();
+        return getNonUniqueAttributeTypesMessage(nonUniqueAttributeTypes);
```

Comment:
This is a great use of SLAP to categorise the type of messages.:smiley:

Reply:
Thanks for the feedback 💯